governance/program/src/processor/process_execute_instruction.rs

```diff
@@ -0,0 +1,80 @@
+//! Program state processor
+
+use borsh::BorshSerialize;
+use solana_program::{
+    account_info::{next_account_info, AccountInfo},
+    clock::Clock,
+    entrypoint::ProgramResult,
+    instruction::Instruction,
+    program::invoke_signed,
+    pubkey::Pubkey,
+    sysvar::Sysvar,
+};
+
+use crate::state::{
+    enums::ProposalState, governance::get_governance_data,
+    proposal::get_proposal_data_for_governance,
+    proposal_instruction::get_proposal_instruction_data_for_proposal,
+};
+
+/// Processes ExecuteInstruction instruction
+pub fn process_execute_instruction(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+
+    let governance_info = next_account_info(account_info_iter)?; // 0
+    let proposal_info = next_account_info(account_info_iter)?; // 0
+    let proposal_instruction_info = next_account_info(account_info_iter)?; // 1
+
+    let clock_info = next_account_info(account_info_iter)?; // 2
+    let clock = Clock::from_account_info(clock_info)?;
+
+    let governance_data = get_governance_data(governance_info)?;
+
+    let mut proposal_data = get_proposal_data_for_governance(proposal_info, governance_info.key)?;
+
+    let mut proposal_instruction_data =
+        get_proposal_instruction_data_for_proposal(proposal_instruction_info, proposal_info.key)?;
+
+    proposal_data.assert_can_execute_instruction(&proposal_instruction_data, clock.slot)?;
+
+    // Execute instruction with Governance PDA as signer
+    let instruction = Instruction::from(&proposal_instruction_data.instruction);
+
+    let instruction_account_infos = account_info_iter.as_slice();
+
+    let mut governance_seeds = governance_data.get_governance_address_seeds()?.to_vec();
+    let (_, bump_seed) = Pubkey::find_program_address(&governance_seeds, program_id);
+    let bump = &[bump_seed];
+    governance_seeds.push(bump);
+
+    invoke_signed(
```

Comment:
We'll need to make it very clear in the documentation that there's no guarantee of instructions happening in any order.  For a v2, we should definitely consider ordered execution of instructions.

Reply:
I'm planning one more follow up PR with improvements which would come from UX testing and I added the ordering requirement to my TODO list. 
It's actually very simple to do since we can assert the instruction being executed has the index equal to the current `instructions_executed_count`

Reply:
Ah right, that is pretty easy.  We'll just have to allow "holes", ie, if I add 10 instructions, then delete 3 of them, then vote and execute, we need to be sure that the "holes" are iterated over correctly.